### PR TITLE
fixes some edge cases in rcd building

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
fix: rcd works when you hit catwalks/lattice
fix: rcd works when you hit foamed floor
/:cl:

### CODE REVIEW

**Comment by TerraGS** [comment]
Something I've wanted but never bothered changing.

File: code/game/objects/structures/lattice.dm [hunk 1]
```diff
@@ -51,6 +51,15 @@
 		new /obj/item/stack/rods(get_turf(src), number_of_rods)
 	qdel(src)
 
+/obj/structure/lattice/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	switch(passed_mode)
```
Comment (ShizCalev) [inline code/game/objects/structures/lattice.dm]: Should only be an `if` if there's only one condition to check.

File: code/game/turfs/simulated/floor/plating.dm [hunk 2]
```diff
@@ -129,6 +129,19 @@
 		else
 			to_chat(user, "<span class='danger'>You hit [src], to no effect!</span>")
 
+/turf/open/floor/plating/foam/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	switch(the_rcd.mode)
+		if(RCD_FLOORWALL)
+			return list("mode" = RCD_FLOORWALL, "delay" = 0, "cost" = 1)
+
+/turf/open/floor/plating/foam/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	switch(passed_mode)
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/plating.dm]: same

File: code/game/turfs/simulated/floor/plating.dm [hunk 3]
```diff
@@ -129,6 +129,19 @@
 		else
 			to_chat(user, "<span class='danger'>You hit [src], to no effect!</span>")
 
+/turf/open/floor/plating/foam/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	switch(the_rcd.mode)
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/plating.dm]: .

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/lattice.dm [hunk 4]
```diff
@@ -51,6 +51,21 @@
...
+	if(the_rcd.mode)
+		if(RCD_FLOORWALL)
+			return list("mode" = RCD_FLOORWALL, "delay" = 0, "cost" = 2)
+
+/obj/structure/lattice/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	if(passed_mode)
+		if(RCD_FLOORWALL)
+			to_chat(user, "<span class='notice'>You build a floor.</span>")
+			var/turf/T = src.loc
+			T.PlaceOnTop(/turf/open/floor/plating)
```
Comment (AnturK) [inline code/game/objects/structures/lattice.dm]: This should validate if it's actually space first.

File: code/game/turfs/simulated/floor/plating.dm [hunk 5]
```diff
@@ -129,6 +129,19 @@
...
+/turf/open/floor/plating/foam/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	if(the_rcd.mode)
+		if(RCD_FLOORWALL)
+			return list("mode" = RCD_FLOORWALL, "delay" = 0, "cost" = 1)
+
+/turf/open/floor/plating/foam/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	if(passed_mode)
+		if(RCD_FLOORWALL)
+			to_chat(user, "<span class='notice'>You build a floor.</span>")
+			PlaceOnTop(/turf/open/floor/plating)
```
Comment (AnturK) [inline code/game/turfs/simulated/floor/plating.dm]: Shouldn't this replace instead ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/game/turfs/simulated/floor/plating.dm (hunk 5 above)
Comment (Tlaltecuhtli) [reply code/game/turfs/simulated/floor/plating.dm]: lattice is a structure and placeontop wont compile without T.

File: code/game/objects/structures/lattice.dm [hunk 6]
```diff
@@ -51,6 +51,22 @@
 		new /obj/item/stack/rods(get_turf(src), number_of_rods)
 	qdel(src)
 
+/obj/structure/lattice/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	if(the_rcd.mode)
+		if(RCD_FLOORWALL)
```
Comment (ShizCalev) [inline code/game/objects/structures/lattice.dm]: if(the_rcd.mode == RCD_FLOORWALL)

File: code/game/objects/structures/lattice.dm [hunk 7]
```diff
@@ -51,6 +51,22 @@
...
+			return list("mode" = RCD_FLOORWALL, "delay" = 0, "cost" = 2)
+
+/obj/structure/lattice/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	if(passed_mode)
+		if(RCD_FLOORWALL)
```
Comment (ShizCalev) [inline code/game/objects/structures/lattice.dm]: .

File: code/game/turfs/simulated/floor/plating.dm [hunk 8]
```diff
@@ -129,6 +129,19 @@
 		else
 			to_chat(user, "<span class='danger'>You hit [src], to no effect!</span>")
 
+/turf/open/floor/plating/foam/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	if(the_rcd.mode)
+		if(RCD_FLOORWALL)
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/plating.dm]: .

File: code/game/turfs/simulated/floor/plating.dm [hunk 9]
```diff
@@ -129,6 +129,19 @@
...
+			return list("mode" = RCD_FLOORWALL, "delay" = 0, "cost" = 1)
+
+/turf/open/floor/plating/foam/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	if(passed_mode)
+		if(RCD_FLOORWALL)
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/plating.dm]: .

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/turfs/simulated/floor/plating.dm [hunk 10]
```diff
@@ -129,6 +129,17 @@
...
 
+/turf/open/floor/plating/foam/rcd_vals(mob/user, obj/item/construction/rcd/the_rcd)
+	if(the_rcd.mode == RCD_FLOORWALL)
+		return list("mode" = RCD_FLOORWALL, "delay" = 0, "cost" = 1)
+
+/turf/open/floor/plating/foam/rcd_act(mob/user, obj/item/construction/rcd/the_rcd, passed_mode)
+	if(passed_mode == RCD_FLOORWALL)
+		to_chat(user, "<span class='notice'>You build a floor.</span>")
+		ChangeTurf(/turf/open/floor/plating)
+			return TRUE
```
Comment (ShizCalev) [inline code/game/turfs/simulated/floor/plating.dm]: extra tab here.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Tlaltecuhtli** [comment]
@AnturK @ShizCalev requests adressed?

**Pull request merged**